spec/services/service_notifications_service_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'rails_helper'
+
+describe ServiceNotificationsService do
+  describe 'notifications' do
+    let(:pom_role) { ['POM'] }
+    let(:spo_role) { ['SPO'] }
+    let(:both_roles) { %w(SPO POM) }
+    let(:filename) { 'app/data/service_notifications.yaml' }
+
+    it 'returns an empty array if file is empty' do
+      allow(YAML).to receive(:load).with(File.read(filename)).and_return('')
+      expect(described_class.notifications(pom_role)).to eq([])
+    end
+
+    it 'returns an empty array if YAML does not return an array' do
+      allow(YAML).to receive(:load).with(File.read(filename)).and_return('{"notifications"=>nil}')
+      expect(described_class.notifications(both_roles)).to eq([])
+    end
+
+    it 'returns an empty array if there are no roles' do
+      roles = []
+      start_date = 2.days.ago.strftime("%d/%m/%Y")
+      data =
+        {
+          "notifications" => [
+            {
+              "id" => "my-new-feature",
+              "start_date" => start_date,
+              "role" => ["POM"],
+              "duration" => 14,
+              "text" => "This is a test banner"
+            }
+          ]
+        }
+
+      allow(YAML).to receive(:load).with(File.read(filename)).and_return(data)
+      expect(described_class.notifications(roles)).to eq([])
+    end
+
+    it 'returns an empty array if there are no notifications that match the users role' do
+      start_date = 4.days.ago.strftime("%d/%m/%Y")
+      data =
+        {
+          "notifications" => [
+            {
+              "id" => "my-new-feature",
+              "start_date" => start_date,
+              "role" => ["POM"],
+              "duration" => 14,
+              "text" => "This is a test banner"
+            }
+          ]
+        }
+
+      allow(YAML).to receive(:load).with(File.read(filename)).and_return(data)
+      expect(described_class.notifications(spo_role)).to eq([])
+    end
+
+    it 'returns an empty array if there are no notifications for today' do
+      start_date = 2.months.ago.strftime("%d/%m/%Y")
+      data =
+        {
+          "notifications" => [
+            {
+              "id" => "my-new-feature",
+              "start_date" => start_date,
+              "role" => ["POM"],
+              "duration" => 30,
+              "text" => "This is a test banner"
+            }
+          ]
+        }
+
+      allow(YAML).to receive(:load).with(File.read(filename)).and_return(data)
```

Comment:
For me this is a code smell. TDD is meant to help with structure. This says that we are welded to the filesystem and YAMl loading. IMHO we should have an iterable of notifications (which can then be passed a YAML.load by someone else) thus making this mock redundant.

Reply:
see above comment about structure - this could be SystemNotification.future (where the future() method is responsible for the filtering)